src/physics/nonlinear_solid.hpp

```diff
@@ -64,6 +64,9 @@ class NonlinearSolid : public BasePhysics {
     // Lame parameters
     double mu;
     double K;
+
+    // Boundary condition information
+    std::unordered_map<std::string, input::BoundaryConditionInputInfo> boundary_conditions;
```

Comment:
What does the string in this map represent?

Reply:
The keys correspond to names of boundary conditions - I'm not sure if that's the best way to disambiguate between them, though.

Reply:
How is this different from the name in `BoundaryConditionInputInfo`?

Reply:
It's duplicated, my thinking was that with name-identified BCs it would be useful to have a map for easier lookup as opposed to just `std::vector<BoundaryConditionInputInfo>`.

It would probably be more natural to allow for string-indexed arrays in Inlet but currently Inlet considers an array to be integer-indexed only.  If that ends up being needed we could add it and then do something like:
```lua
boundary_conds = {
   ["displacement"] = { attrs = {1}, },
   ["traction"] = { attrs = {2}, },
}
```

Reply:
I think the string index is a good thing to work toward, but for now I think reducing the duplication makes sense for clarity. I can imagine each physics module looping through the boundary condition input and setting them up in a  like this:
```cpp
for (const auto& bc : boundary_conditions) {
  if (bc.name == "displacement") {
    // add a displacement bc...
  } else if (bc.name == "traction") {
    // add a traction bc...
  }
}